build.gradle

```
@@ -134,24 +134,18 @@ import java.util.regex.Matcher
 import java.util.regex.Pattern
 import org.jsoup.Jsoup
 import org.jsoup.nodes.Element
+import org.jsoup.select.Elements
 import org.jsoup.nodes.Document
 
-String findNewestEapBuildNumber() {
+String checkForEapWithBuildNumberHigherThan(String latestEapBuildNumber) {
   Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
-  Element link = htmlContent.select('a[href~=' + project.intellijSnapshotsUrl + 'com/jetbrains/intellij/idea/ideaIC/\\d+\\.\\d+\\.\\d+][href$="-EAP-SNAPSHOT.pom"]')[0]
-  if (link) {
+  Elements links = htmlContent.select('a[href~=' + project.intellijSnapshotsUrl + 'com/jetbrains/intellij/idea/ideaIC/\\d+\\.\\d+\\.\\d+][href$="-EAP-SNAPSHOT.pom"]')
```

Nit: coz you now do the regex filtering on each individual link anyway, this can be probably reduced/simplified to:
```suggestion
  Elements links = htmlContent.select('a[href^=' + project.intellijSnapshotsUrl + 'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')
```

---

Albeit it's a matter of taste, up to you